packages/editor/src/dataviews/store/private-actions.ts

```diff
@@ -139,3 +177,46 @@ export const registerPostTypeActions =
 
 		doAction( 'core.registerPostTypeActions', postType );
 	};
+
+export const registerPostTypeFields =
+	( postType: string ) =>
+	async ( { registry }: { registry: any } ) => {
+		const isReady = unlock( registry.select( editorStore ) ).isEntityReady(
+			'postType',
+			postType,
+			'fields'
+		);
+		if ( isReady ) {
+			return;
+		}
+
+		unlock( registry.dispatch( editorStore ) ).setIsReady(
+			'postType',
+			postType,
+			'fields'
+		);
+
+		const fields = [
+			featuredImageField,
```

Comment:
Don't we need the post type config to add some fields conditionally? For example featured image is the post type supports it, etc..

Reply:
Yes! But I don't want to do it in this PR, I'd rather work on small steps that land fast :)

There's a few other things we need to do: for example, somewhere we need to keep a list of post types <=> fields ([see related conversation](https://github.com/WordPress/gutenberg/pull/67175/files#r1850731074)), but also a bit of validation (return early if post type is unknown), etc..

Would that work for you?